resources/shared/params.mjs

```diff
@@ -193,6 +205,14 @@ export class Params {
     }
 }
 
+function isValidJsonUrl(url) {
+    try {
+        return new URL(url) && url.toLowerCase().endsWith(".json");
```

Comment:
This will return true for `foo.html?query=a.json` or `foo.html#bar.json`.
I don't think that's what we want? Can we just use `URL`'s `pathname` instead?